exporter/awsemfexporter/README.md

```diff
@@ -64,6 +64,7 @@ A metric descriptor section allows the schema of a metric to be overwritten befo
 | `unit` | The overwritten value of unit. The [MetricDatum](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html) contains a ful list of supported unit values. |         |
 | `overwrite` | `true` if the schema should be overwritten with the given specification, otherwise it will only be configured if empty. |   false   |
 
+**Note:** Under cloudwatch when you declare metrics on the log event within "Metrics" section - this section can have up to 100 metric definition objects which means that you can have up to 100 metrics defined in that section. This array **MUST NOT** contain more than 100 MetricDefinition objects.
```

Comment:
It's not clear to me that this is attached to the correct configuration entity or that this is the correct recommendation.  It seems from the report that the count of `MetricDefinition` objects is the limiting factor and that count varies based on the number of metrics processed in a batch and not the number of elements in the `metric_descriptor` configuration:

> We then sent metric payloads of 250+ metrics through the pipeline every ~30s, and noticed that the resulting log events received in Amazon CloudWatch contained a number of metrics much less than 100 for each payload (for example, batches of 20-50 metrics per log event).

Do we instead need to be recommending the use of the `batch` processor and that the customer adjust maximum batch size until they are reliably below these limits if they encounter the limits?  Can we emit a warning to the logs if we detect that this limit is reached?

Reply:
Another thing to consider would be: would be more better changing the batch when the current batch is full of 100 Metric Definition. In that way, we can avoid customers overlimit the MetricDefinition instead of only warning logs.